classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.inc.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I8073_RemoveNotesWithoutQueriesAndRelatedObjects.php
+ *
+ * Copyright (c) 2014-2022 Simon Fraser University
+ * Copyright (c) 2000-2022 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I7706_AssociateTemplatesWithMailables
+ * @brief Refactors relationship between Mailables and Email Templates
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use PKP\migration\Migration;
+use Illuminate\Support\Facades\DB;
+use PKP\submissionFile\SubmissionFile;
+use APP\core\Services;
+use PKP\install\DowngradeNotSupportedException;
+use PKP\config\Config;
+use League\Flysystem\Local\LocalFilesystemAdapter;
+use League\Flysystem\UnixVisibility\PortableVisibilityConverter;
+use PKP\file\FileManager;
+use League\Flysystem\Filesystem;
+use League\Flysystem\FilesystemException;
+use League\Flysystem\UnableToDeleteFile;
+
+class I8073_RemoveNotesWithoutQueriesAndRelatedObjects extends Migration
+{
+    private const ASSOC_TYPE_NOTE = 0x0000208; // PKPApplication::ASSOC_TYPE_NOTE
+    private const ASSOC_TYPE_QUERY = 0x010000a; // PKPApplication::ASSOC_TYPE_QUERY
+    private const SUBMISSION_FILE_QUERY = 18; // SubmissionFile::SUBMISSION_FILE_QUERY
+    private const FILE_MODE_MASK = 0666; // FileManager::FILE_MODE_MASK
+    private const DIRECTORY_MODE_MASK = 0777; // FileManager::DIRECTORY_MODE_MASK
+
+    public function up(): void
+    {
+        $orphanedIds = DB::table('notes AS n')
+            ->leftJoin('queries AS q', 'n.assoc_id', '=', 'q.query_id')
+            ->where('n.assoc_type', '=', self::ASSOC_TYPE_QUERY)
+            ->whereNull('q.query_id')
+            ->pluck('n.note_id', 'n.assoc_id');
+
+        foreach ($orphanedIds as $neQueryId => $noteId) {
+            error_log("Removing submission files that relates to the note entry ID ${noteId} which will be deleted as orphan object");
+            $notesFileRows = DB::table('submission_files as sf')
+                ->join('files as f', 'sf.file_id', '=', 'f.file_id')
+                ->where('sf.assoc_type', '=', self::ASSOC_TYPE_NOTE)
+                ->where('sf.assoc_id', '=', $noteId)
+                ->where('sf.file_stage', '=', self::SUBMISSION_FILE_QUERY)
+                ->get([
+                        'sf.submission_file_id as submissionFileId',
+                        'sf.file_id as fileId',
+                        'f.path as filePath'
+                    ]);
+
+            foreach ($notesFileRows as $submissionFileRow) {
+                $submissionFileId = $submissionFileRow->submissionFileId;
+                $submissionFileFileId = $submissionFileRow->fileId;
+                $submissionFilePath = $submissionFileRow->filePath;
+
+                error_log("Removing submission file revisions that relate to the submission file entry ID ${submissionFileId} which will be deleted.");
```

Comment:
These `error_log` lines will create a lot of noise. I'd suggest checking the return value of `delete` and only adding a log entry if something was actually removed.

Reply:
Fixed - Added the number of deleted items to the message also.